### PR TITLE
Refactor operations.py into class

### PR DESCRIPTION
**Pull request by LJBabbage**
### What is the context of this PR?
Make operations into a class, and move instantiated higher. The instantiation has been moved higher to allow access to schema etc, but at present that is not implemented and on a different card

### How to review 
Make sure everything still works

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: tests/app/questionnaire/routing/test_operators.py [hunk 1]
```diff
@@ -343,5 +369,27 @@ def test_nonetype_operands_for_array_operators(operator_name, operands):
     ],
 )
 def test_operation_count(operands, expected_result):
-    operator = Operator(Operator.COUNT)
+    operation = get_operation(Operator.COUNT)
+    operator = Operator(Operator.COUNT, operation)
     assert operator.evaluate(operands) is expected_result
+
+
+def get_operation(operator):
```
Comment (LJBabbage) [inline tests/app/questionnaire/routing/test_operators.py:377]: This is a pain, but as operations will eventually be initialized with schema, locale etc and the function names don't match up I have had to put a map here, however I realise this is duplicated in when_rule_evaluator.

File: app/questionnaire/routing/when_rule_evaluator.py [hunk 2]
```diff
@@ -36,12 +37,30 @@ def __post_init__(self) -> None:
...
 
     def _evaluate(self, rule: dict[str, Sequence]) -> Union[bool, Optional[date]]:
-        operator = Operator(next(iter(rule)))
-        operands = rule[operator.name]
+        next_rule = next(iter(rule))
+        operator = Operator(next_rule, self.operation_mapping[next_rule])
+        operands = rule[next_rule]
 
-        if not isinstance(operands, Sequence):
+        if not isinstance(rule[next_rule], Sequence):
```
Comment (petechd) [inline app/questionnaire/routing/when_rule_evaluator.py:63]: I think you forgot about using previously assigned `operands` here?

Comment (LJBabbage) [reply app/questionnaire/routing/when_rule_evaluator.py:63]: Oh yeah, I changed it, then realised it was used elsewhere, but must of missed changing it back, thanks

**Review by berroar — APPROVED** [review]
LGTM beyond the comments above 👍

**Review by petechd — APPROVED** [review]
Looks good!

**Pull request merged**